xbmc/rendering/dx/DeviceResources.cpp

```diff
@@ -1097,3 +1112,159 @@ void DX::DeviceResources::Trim() const
+          hdrCapable = true;
+          constexpr double FACTOR_1 = 50000.0;
+          constexpr double FACTOR_2 = 10000.0;
+          hdr10.RedPrimary[0] = static_cast<uint16_t>(FACTOR_1 * od.RedPrimary[0]);
+          hdr10.RedPrimary[1] = static_cast<uint16_t>(FACTOR_1 * od.RedPrimary[1]);
+          hdr10.GreenPrimary[0] = static_cast<uint16_t>(FACTOR_1 * od.GreenPrimary[0]);
+          hdr10.GreenPrimary[1] = static_cast<uint16_t>(FACTOR_1 * od.GreenPrimary[1]);
+          hdr10.BluePrimary[0] = static_cast<uint16_t>(FACTOR_1 * od.BluePrimary[0]);
+          hdr10.BluePrimary[1] = static_cast<uint16_t>(FACTOR_1 * od.BluePrimary[1]);
+          hdr10.WhitePoint[0] = static_cast<uint16_t>(FACTOR_1 * od.WhitePoint[0]);
+          hdr10.WhitePoint[1] = static_cast<uint16_t>(FACTOR_1 * od.WhitePoint[1]);
+          hdr10.MaxMasteringLuminance = static_cast<uint32_t>(FACTOR_2 * od.MaxLuminance);
+          hdr10.MinMasteringLuminance = static_cast<uint32_t>(FACTOR_2 * od.MinLuminance);
+          hdr10.MaxContentLightLevel = static_cast<uint16_t>(od.MaxFullFrameLuminance);
+          hdr10.MaxFrameAverageLightLevel = static_cast<uint16_t>(od.MaxFullFrameLuminance);
+        }
+        else if (od.MaxLuminance >= 400.0)
+        {
+          CLog::LogF(LOGNOTICE, "Monitor HDR capable is detected but NOT enabled:");
+          hdrCapable = true;
+        }
+        else
+        {
+          CLog::LogF(LOGNOTICE, "No monitor HDR capable detected.");
+        }
+        if (hdrCapable)
+        {
+          std::string txColorSpace = "UNKNOWN";
+          switch (od.ColorSpace)
+          {
+            case DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020:
+              txColorSpace = "DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020";
+              break;
+            case DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709:
+              txColorSpace = "DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709";
+              break;
+          }
+          CLog::LogF(LOGNOTICE, "ColorSpace = {0:s}", txColorSpace);
+          CLog::LogF(LOGNOTICE, "RedPrimary = {0:0.3f}, {1:0.3f}", od.RedPrimary[0],
+                     od.RedPrimary[1]);
+          CLog::LogF(LOGNOTICE, "GreenPrimary = {0:0.3f}, {1:0.3f}", od.GreenPrimary[0],
+                     od.GreenPrimary[1]);
+          CLog::LogF(LOGNOTICE, "BluePrimary = {0:0.3f}, {1:0.3f}", od.BluePrimary[0],
+                     od.BluePrimary[1]);
+          CLog::LogF(LOGNOTICE, "WhitePoint = {0:0.3f}, {1:0.3f}", od.WhitePoint[0],
+                     od.WhitePoint[1]);
+          CLog::LogF(LOGNOTICE, "MinLuminance = {0:0.4f}", od.MinLuminance);
+          CLog::LogF(LOGNOTICE, "MaxLuminance = {0:0.0f}", od.MaxLuminance);
+          CLog::LogF(LOGNOTICE, "MaxFullFrameLuminance = {0:0.0f}", od.MaxFullFrameLuminance);
+        }
+      }
+      else
+      {
+        CLog::LogF(LOGERROR, "DXGI GetDesc1 failed");
+      }
+    }
+  }
+  //Saves Monitor parameters for use later
+  m_displayHDR10 = hdr10;
+  if (HDRCapable != nullptr)
```

Comment:
Drop this completely? HDRCapable is always only nullptr, as its the only thing ever passed into the method parameter call

Reply:
Is a check for optional pointer.

Reply:
Same as above. Better to bring it in when an actual functional commit makes use of it

Reply:
Done